SETUP.md

```diff
@@ -0,0 +1,121 @@
+# Setup Guide
+
+This document describes how to setup all the dependencies to run the notebooks
+in this repository.
+
+Many computer visions scenarios are extremely computationlly heavy. Training a
+model often requires a machine that has a GPU, and would otherwise be too slow.
+We recommend using the GPU-enabled [Azure Data Science Virtual Machine (DSVM)](https://azure.microsoft.com/en-us/services/virtual-machines/data-science-virtual-machines/) since it comes prepared with a lot of the prerequisites needed to efficiently do computer vision.
+
+To scale up or to operationalize your models, we recommend setting up [Azure
+ML](https://docs.microsoft.com/en-us/azure/machine-learning/). Our notebooks
+provide instructions on how to use it.
+
+
+## Table of Contents
+
+1. [Compute Environment](#compute-environments)
+1. [System Requirements](#system-requirements)
+1. [Installation](#installation)
+1. [Tunneling](#tunneling)
+
+## Compute Environments
+
+Most computer vision scenarios require a GPU, especially if you're training a
+custom model. We recommend using a virtual machine to run the notebooks on.
+Specifically, we'll want one with a powerful GPU. The NVIDIA's Tesla V100 is a
+good choice that can be found in most Azure regions.
+
+The easiest way to get started is to use the [Azure Data Science Virtual Machine (DSVM)](https://azure.microsoft.com/en-us/services/virtual-machines/data-science-virtual-machines/). This VM will come installed with all the system requirements that are needed to run the notebooks in this repository. If you choose this option, you can skip the [System Requirements](#system-requirements) step in this guide as those requirements come pre-installed on the DSVM.
+
+Here are some ways you can create the DSVM:
+
+__Virtual Machine Builder__
+
+One easy way to create your DSVM is to use the [VM Builder](../contrib/vm_builder) tool located inside of the 'contrib' folder in the root directory of the repo. Simply run `python contrib/vm_builder/vm_builder.py` at the root level of the repo and this tool will preconfigure your virtual machine with the appropriate settings for working with this repository.
```

Comment:
This section should probably be moved until after the "provision a data science VM.." section and pointed out that our recommendation is to use the portal. That's partly since the portal is easy to use and I think good for beginners to know what exactly they are creating and how to delete it. And partly since the tool could stop working (since we don't test it).